icu4c/source/common/umutex.cpp

```diff
@@ -99,13 +73,13 @@ umtx_condSignal(UConditionVar *cond) {
  *************************************************************************************************/
 
 static std::mutex &initMutex() {
-    static std::mutex m;
-    return m;
+    static std::mutex *m = STATIC_NEW(std::mutex);
+    return *m;
```

Comment:
What happens when the ICU binary gets unloaded? Does this content leak? And if it does, what exactly leaks? Is std::mutex backed by an OS resource such as a critical section that then persists for the lifetime of the process? If so, that needs to be avoided.

Reply:
I think the implementation details of std::mutex might be compiler and/or library dependent.
If it is backed by a Critical Section, then we would need some way to invoke the destructor for this object.

Reply:
I looked it up, and the Windows inbox version of the CRT uses critical sections. That means if you're constructing this object you need to make sure it is destructed on DLL exit.

In general I would recommend considering forking the behavior here. There are scenarios where init/uninit is possible in DLL attach/detach is possible and desirable. Those should take advantage of that, and avoid the overhead and correctness issues that come with dynamic usage.

---
In reply to: [279541894](https://github.com/unicode-org/icu/pull/620#discussion_r279541894) [](ancestors = 279541894)